### DISCUSSION TITLE
Make the RSD embeddable

### DISCUSSION THREAD

**Discussion by ewan-escience** [research-software-directory/RSD-as-a-service#250] [Ideas]
There was the idea to make the RSD embeddable in other websites, see also this image:
![draft of making the RSD embeddable](https://user-images.githubusercontent.com/4195550/166263884-328af974-ce50-4640-a265-c7812a8db9e4.png)

Let's discuss this idea here. Some considerations are:
* How should it be made embeddable, e.g. through an `<iframe>` only?
* What parts of the website to include to make it embeddable?
* How to request the embeddable version, e.g. with a query parameter `embed=true`?
* What technical adaptations have to be made?
* How to make navigation work?
* How much work will this take?

**Comment by ctwhome** [comment]
Probably the easiest way to achieve this is by being able to hide and show specific parts of the website by a given param in the URL when embedding into an iFrame. See draft PR #256 

By doing this, we don't have to develop platform-specific plugins, for instance, a plugin for WordPress, but just embedding the website software pages or projects, or any other public page.

Another solution is creating custom frontends retrieving the content from the official RSD API.

**Comment by cmeessen** [comment]
We currently favour the solution using the official RSD API. An idea was to create a basic JavaScript library that can be used to fetch the data and that will build a basic DOM tree.

Will the RSD API stay open, or are you planning to restrict access from given URLs?

**Reply by ctwhome** [reply]
As far as I know, It will be open. I imagine that we should discuss how to protect the API to avoid misuse, but in principle, it should stay open.

**Reply by ewan-escience** [reply]
The current plan is to keep it open indeed. If necessary, we can configure NGINX to use [rate limiting](https://www.nginx.com/blog/rate-limiting-nginx/).
